### DISCUSSION TITLE
Add bulks of elements as JSON

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1778] [Q&A] [answered]
We can use graph.fromJSON to add a bulk of cells at once as JSON. However, this clears the graph everytime. In my case, I want to add batches of operations; each batch as JSON. However, each batch clears the previous once.

Which API would be recommended in my case?

**Comment by kumilingus** [comment, marked as answer]
The `dia.Graph.prototype.addCells` method also accepts an array of JSON cells.

```js
graph.addCells([{
  type: 'standard.Rectangle',
  size: { width: 100, height: 100 }
}, {
  type: 'standard.Ellipse',
  size: { width: 200, height: 100 }
}]);
```
Instead of ` graph.fromJSON(json)`,  the following can be called to preserve existing cells:

```js
graph.addCells(json.cells);
```

**Reply by seekeramento** [reply]
Thanks
